src/node/src/node.rs

```diff
@@ -182,44 +182,32 @@ impl QueryResultMapper {
             }
             Ok(QueryEvent::RecordsUpdated(records)) => vec![Message::CommandComplete(format!("UPDATE {}", records))],
             Ok(QueryEvent::RecordsDeleted(records)) => vec![Message::CommandComplete(format!("DELETE {}", records))],
-            Err(QueryError::SchemaAlreadyExists(schema_name)) => vec![Message::ErrorResponse(
-                Some("ERROR".to_owned()),
-                Some("42P06".to_owned()),
-                Some(format!("schema \"{}\" already exists", schema_name)),
-            )],
-            Err(QueryError::SchemaDoesNotExist(schema_name)) => vec![Message::ErrorResponse(
-                Some("ERROR".to_owned()),
-                Some("3F000".to_owned()),
-                Some(format!("schema \"{}\" does not exist", schema_name)),
-            )],
-            Err(QueryError::TableAlreadyExists(table_name)) => vec![Message::ErrorResponse(
-                Some("ERROR".to_owned()),
-                Some("42P07".to_owned()),
-                Some(format!("table \"{}\" already exists", table_name)),
-            )],
-            Err(QueryError::TableDoesNotExist(table_name)) => vec![Message::ErrorResponse(
-                Some("ERROR".to_owned()),
-                Some("42P01".to_owned()),
-                Some(format!("table \"{}\" does not exist", table_name)),
-            )],
-            Err(QueryError::ColumnDoesNotExist(non_existing_columns)) => {
-                let error_message = if non_existing_columns.len() > 1 {
-                    format!("columns {} do not exist", non_existing_columns.join(", "))
-                } else {
-                    format!("column {} does not exist", non_existing_columns[0])
+            Err(QueryError { severity, code, kind }) => {
```

Comment:
I think it would be better if this code will be moved to
```rust
impl Display for QueryError
```
so then we could do the following
```rust
match resp {
// ...
    Err(error) => vec![Message::ErrorResponse(Some(error.severity.into()), Some(error.code), Some(format!("{}", error)))]
```
In this case `QueryError` is responsible for message formatting.
I think we can start improving code base with step-by-step encapsulation so that easier to move code around (it isn't much ergonomic right now 😞 ) and message formatting is a good candidate.
I also believe it will help with #129

Reply:
Let also add `fn code(&self) -> Option<String>` and `fn severity(&self) -> Option<String>` to `QueryError`. I am not sure but that should help make `QueryErrorKind` and `Severity` `enum`s have `pub (crate)` visibility.

Reply:
Alright, sounds good.

Reply:
Is there a reason for making `fn code(&self))` return an option as opposed to return an `&str` or `String`?

Reply:
Well, PostgreSQL documentation says that code is optional part of the message, and thought that it be nicer to create `Message::ErrorResponse` just like
```rust
Message::ErrorResponse(error.severity(), error.code(), format!("{}", error));
```